src/org/traccar/database/DataManager.java

```diff
@@ -115,6 +124,73 @@ private void initDatabase() throws Exception {
         }
     }
 
+    public static String constructObjectQuery(String action, Class<?> clazz, boolean additional) {
+        switch (action) {
+            case ACTION_INSERT:
+            case ACTION_UPDATE:
+                StringBuilder result = new StringBuilder();
+                StringBuilder fields = new StringBuilder();
+                StringBuilder values = new StringBuilder();
+
+                Set<Method> methods = new HashSet<>(Arrays.asList(clazz.getMethods()));
+                methods.removeAll(Arrays.asList(Object.class.getMethods()));
+                methods.removeAll(Arrays.asList(BaseModel.class.getMethods()));
+                for (Method method : methods) {
+                    if (method.getName().startsWith("get") && method.getParameterTypes().length == 0
+                            && (additional ? method.isAnnotationPresent(QueryAdditional.class)
+                                : !method.isAnnotationPresent(QueryIgnore.class)
+                                && !method.isAnnotationPresent(QueryAdditional.class))) {
+                        String name = method.getName().substring(3, 4).toLowerCase()
```

Comment:
I think you can use `Introspector.decapitalize` here... maybe `Introspector` can be used in some other places as well (managers?).